test/features/lists.js

```diff
@@ -0,0 +1,1270 @@
+'use strict';
+
+const nock = require('nock');
+const assert = require('../utils/assert.js');
+const server = require('../utils/server.js');
+const querystring = require('querystring');
+const preq = require('preq');
+
+
+describe('reading lists', function() {
+    const csrfToken = '<mock>';
+    const sessionCookies = '<mock>';
+    const config = {
+        // domain: 'dev.wiki.local.wmftest.net',
+        // baseURL: 'http://localhost:7231/dev.wiki.local.wmftest.net/v1',
+        baseURL: server.config.baseURL,
+        apiDomain: server.config.apiURL.replace(/^(.*)(\/w\/.+)$/, "$1"),
+        apiPath: server.config.apiURL.replace(/^(.*)(\/w\/.+)$/, "$2"),
+    };
+
+    function getApi() {
+        return nock(config.apiDomain, {
+            reqheaders: {
+                'Cookie': sessionCookies,
+                'Host': config.domain,
+            },
+        })
+        .defaultReplyHeaders({
+            'Content-Type': 'application/json; charset=utf-8',
+        });
+    }
+
+    // workaround for https://github.com/node-nock/nock/issues/852
+    // also for object ordering issues
+    function nockDiff(expected) {
+        return (actual) => {
+            if (typeof actual === 'string' && typeof expected === 'object') {
+                actual = querystring.parse(actual);
+            }
+            try {
+                assert.deepEqual(actual, expected, 'nock failure');
+                return true;
+            } catch (e) {
+                console.log(e);
+                return false;
+            }
+        }
+    }
+
+    this.timeout(20000);
+
+    before(function() {
+        //nock.recorder.rec();
+        return server.start()
+        // Do a preparation request to force siteinfo fetch so that we don't need to mock it
+        .then(() => preq.get({
+            uri: `${config.baseURL}/page/html/Main_Page`,
+        }))
+        // After this, there should be no unmocked requests other than those to /lists
+        .then(() => nock.emitter.on('no match', req => {
+            if (!req.href.startsWith(`${config.baseURL}/data/lists/`)) {
+                throw Error(`Unmocked request to ${req.href}`);
+            }
+        }));
```

Comment:
I'm worried this will be thrown out of the test context and might not be handled correctly and will basically crash the whole suite. Crashing the suite is not a big deal, failure is failure, but I'm wondering whether this message will be properly printed